### PR TITLE
Clean up testgrid dashboard names and groups

### PR DESCRIPTION
**Pull request by spiffxp**
See commits for details

ref: https://github.com/kubernetes/test-infra/issues/9363

### CODE REVIEW

**Comment by spiffxp** [comment]
/uncc @stevekuznetsov @ixdy 
/cc @krzyzacy @cjwagner @jberkus 
/kind cleanup
FYI @kubernetes/sig-api-machinery-pr-reviews @kubernetes/sig-scalability-pr-reviews

**Comment by spiffxp** [comment]
...sigh, wrong button
also FYI @kubernetes/sig-gcp-pr-reviews and finally @fejta (for the google-rules_k8s move)

**Comment by krzyzacy** [comment]
/lgtm

hummm missed those... I think you can add the stable-foo dependent dashboards to https://github.com/kubernetes/test-infra/blob/master/experiment/fix_testgrid_config.py#L43-L50

I'll improve that logic a bit for 1.13

**Comment by spiffxp** [comment]
/hold
I'll update that script and see what happens

File: testgrid/config.yaml [hunk 1]
```diff
@@ -4165,9 +4165,9 @@ dashboards:
     test_group_name: ci-kubernetes-e2e-gci-gce-reboot
   - name: gci-gce-scalability
     test_group_name: ci-kubernetes-e2e-gci-gce-scalability
-  - name: gci-gce-scalability-1.10
+  - name: gci-gce-scalability-1.11
```
[the remark this answers is not in the document]
Comment (cjwagner) [reply testgrid/config.yaml]: Ah, looks like that is addressed here: https://github.com/kubernetes/test-infra/pull/9712#issuecomment-427447400

[the remark this answers is not in the document]
Comment (krzyzacy) [reply testgrid/config.yaml]: these dashboards were not added to the list - I believe Aaron just added more dashboards and rerun the script.

[the remark this answers is not in the document]
Comment (spiffxp) [reply testgrid/config.yaml]: the script was missing a few dashboards, which I just added in my last commit

**Comment by spiffxp** [comment]
/hold cancel
I'll send out a pardon-our-dust e-mail when this merges and looks as expected.  I doubt this is the last naming change we're going to make as we look to resolve release dashboard job and naming consistency

**Comment by spiffxp** [comment]
pardon our dust sent: https://groups.google.com/d/msg/kubernetes-dev/pWzo04h63V8/ieBYZbf8AAAJ

**Pull request merged**